cscs-checks/prgenv/openacc_checks.py

```diff
@@ -1,50 +1,46 @@
-import os
-
+import reframe as rfm
 import reframe.utility.sanity as sn
-from reframe.core.pipeline import RegressionTest
 
-class OpenACCFortranCheck(RegressionTest):
-    def __init__(self, num_tasks, **kwargs):
+
+@rfm.parameterized_test([1], [2])
+class OpenACCFortranCheck(rfm.RegressionTest):
+    def __init__(self, num_tasks):
         if num_tasks == 1:
-            check_name = 'openacc_fortran_check'
+            self.name = 'openacc_fortran_check'
         else:
-            check_name = 'openacc_mpi_fortran_check'
-        super().__init__(check_name,
-                         os.path.dirname(__file__), **kwargs)
+            self.name = 'openacc_mpi_fortran_check'
+
+        super().__init__()
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
         if self.current_system.name in ['daint', 'dom']:
             self.modules = ['craype-accel-nvidia60']
-            self._pgi_flags = '-acc -ta=tesla:cc60'
+            self._pgi_flags = ['-acc', '-ta=tesla:cc60']
         elif self.current_system.name in ['kesch']:
             self.modules = ['craype-accel-nvidia35']
-            self._pgi_flags = '-acc -ta=tesla:cc35'
+            self._pgi_flags = ['-acc', '-ta=tesla:cc35']
```

Comment:
I would put this `if/else` inside the `setup()` method, specifically for the PGI environment.